src/main/kotlin/uk/gov/justice/digital/hmpps/hmppsinterventionsservice/service/CommunityAPIOffenderService.kt

```diff
@@ -86,4 +104,24 @@ class CommunityAPIOffenderService(
       .block()
       ?.staffIdentifier
   }
+
+  fun getResponsibleOfficer(crn: String): ResponsibleOfficer {
+    val offenderManagersPath = UriComponentsBuilder.fromPath(offenderManagersLocation)
+      .buildAndExpand(crn)
+      .toString()
+
+    // technically this endpoint can return more than one offender manager that
+    // is a responsible officer. in practice this should not happen. if it does
+    // we just take the first one we come across.
+    return communityApiClient.get(offenderManagersPath)
```

Comment:
Possibly could order it by `fromDate` on `CommunityOrPrisonOffenderManager` and get the latest that is set in the past.

It seems this is populated from `ALLOCATION_DATE` from `OFFENDER_MANAGER` table. 

The `END_DATE` is skipped in population of the POJO, so maybe can make a change to include it to also filter out "ended" responsibile officers.

Reply:
i don't think this is quite accurate - the response from 'allOffenderManagers' first filters OM's by `endDate == null && Optional.ofNullable(activeFlag).orElse(0L) == 1L && !isDeleted()` (`OffenderManager::isActive`), then populates 'isResponsibleOfficer' by checking `endDateTime == null` (`ResponsibleOfficer::isActive`). so ended OMs and ROs are not included. 

whilst there are no DB constraints for ensuring there cannot be more than one active RO at any given time, the only way to update ROs via the API ensure that previous ROs have end dates set. so i am totally happy that this method is sufficient without adding additional complexity.

worst case that there are multiple active ROs isn't that impactful, since we still select one of them.

Reply:
I understand now, thank you